cartoframes/context.py

```diff
@@ -506,33 +446,115 @@ def fetch(self, query, decode_geom=False):
         """
         copy_query = 'COPY ({query}) TO stdout WITH (FORMAT csv, HEADER true)'.format(query=query)
         query_columns = get_columns(self, query)
+
         result = recursive_read(self, copy_query)
-        df = pd.read_csv(result)
+        df_types = dtypes(query_columns, exclude_dates=True)
+
+        df = pd.read_csv(result, dtype=df_types,
+                         parse_dates=date_columns_names(query_columns),
+                         true_values=['t'],
+                         false_values=['f'],
+                         index_col='cartodb_id' if 'cartodb_id' in df_types.keys() else False,
+                         converters={'the_geom': lambda x: _decode_geom(x) if decode_geom else x})
+
+        if decode_geom:
+            df.rename({'the_geom': 'geometry'}, axis='columns', inplace=True)
+
+        return df
+
+    def execute(self, query):
+        """Runs an arbitrary query to a CARTO account.
+
+           This method is specially useful for queries that do not return any data and just
+           perform a database operation like:
+
+             - INSERT, UPDATE, DROP, CREATE, ALTER, stored procedures, etc.
+
+           Queries are run using a `Batch SQL API job
+           <https://carto.com/developers/sql-api/guides/batch-queries/>`__
+           in the user account
+
+           The execution of the queries is asynchronous but this method automatically
+           waits for its completion (or failure). The `job_id` of the Batch SQL API job
+           will be printed. In case there's any issue you can contact the CARTO support team
+           specifying that `job_id`.
+
+        Args:
+            query (str): An SQL query to run against CARTO user database.
+
+        Returns:
+            None
+
+        Raises:
+            CartoException: If the query fails to execute
+
+        Examples:
+
+            Drops `my_table`
+
+            .. code:: python
+
+                cc.execute(
+                    '''
+                      DROP TABLE my_table
+                    '''
+                )
+
+            Updates the column `my_column` in the table `my_table`
+
+            .. code:: python
+
+                cc.query(
+                    '''
+                      UPDATE my_table SET my_column = 1
+                    '''
+                )
+
+        """
+        self.batch_sql_client.create_and_wait_for_completion(query)
+
+    def query(self, query, table_name=None, decode_geom=False, is_select=True):
```

Comment:
I'd try to do some magic with the `is_select` parameter. Something like default it to `None` and then check if it is None and do some auto-detection (can be something as simple as a `query.lower().startswith('select')`. This way we keep backwards compatibility a bit better.

Reply:
I see what you mean... with the current solution if you want to do an `UPDATE` for example, the method signature changes because you have to call it with `is_select=False`...

Honestly if we do `auto-detection` on the query parameter I'd prefer to do anything more accurate than that, but probably, it is good enough by now

Reply:
Yeah, `auto-detection`:tm: is going to suck anyway. We are never going to get it right 100%, so the parameter should stay (to be able to override). But if we catch 90% of our cases, I think it's a win to be convenient. cc @andy-esch since he probably knows way more than we do about the user-facing part of this.